clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignorAdapter.java

```diff
@@ -0,0 +1,136 @@
+    private final PartitionAssignor oldAssignor;
+
+    PartitionAssignorAdapter(PartitionAssignor oldAssignor) {
+        this.oldAssignor = oldAssignor;
+    }
+
+    @Override
+    public ByteBuffer subscriptionUserData(Set<String> topics) {
+        return oldAssignor.subscription(topics).userData();
+    }
+
+    @Override
+    public GroupAssignment assign(Cluster metadata, GroupSubscription groupSubscription) {
+        return toNewGroupAssignment(oldAssignor.assign(metadata, toOldGroupSubscription(groupSubscription)));
+    }
+
+    @Override
+    public void onAssignment(Assignment assignment, ConsumerGroupMetadata metadata) {
+        oldAssignor.onAssignment(toOldAssignment(assignment), metadata.generationId());
+    }
+
+    @Override
+    public String name() {
+        return oldAssignor.name();
+    }
+
+    private static PartitionAssignor.Assignment toOldAssignment(Assignment newAssignment) {
+        return new PartitionAssignor.Assignment(newAssignment.partitions(), newAssignment.userData());
+    }
+
+    private Map<String, PartitionAssignor.Subscription> toOldGroupSubscription(GroupSubscription newSubscriptions) {
+        Map<String, PartitionAssignor.Subscription> oldSubscriptions = new HashMap<>();
+        for (Map.Entry<String, Subscription> entry : newSubscriptions.groupSubscription().entrySet()) {
+            String member = entry.getKey();
+            Subscription newSubscription = entry.getValue();
+            oldSubscriptions.put(member, new PartitionAssignor.Subscription(
+                newSubscription.topics(), newSubscription.userData()));
+        }
+        return oldSubscriptions;
+    }
+
+    private GroupAssignment toNewGroupAssignment(Map<String, PartitionAssignor.Assignment> oldAssignments) {
+        Map<String, Assignment> newAssignments = new HashMap<>();
+        for (Map.Entry<String, PartitionAssignor.Assignment> entry : oldAssignments.entrySet()) {
+            String member = entry.getKey();
+            PartitionAssignor.Assignment oldAssignment = entry.getValue();
+            newAssignments.put(member, new Assignment(oldAssignment.partitions(), oldAssignment.userData()));
+        }
+        return new GroupAssignment(newAssignments);
+    }
+
+    /**
+     * Get a list of configured instances of {@link org.apache.kafka.clients.consumer.ConsumerPartitionAssignor}
+     * based on the class names/types specified by {@link org.apache.kafka.clients.consumer.ConsumerConfig#PARTITION_ASSIGNMENT_STRATEGY_CONFIG}
+     * where any instances of the old {@link PartitionAssignor} interface are wrapped in an adapter to the new
+     * {@link org.apache.kafka.clients.consumer.ConsumerPartitionAssignor} interface
+     */
+    public static List<ConsumerPartitionAssignor> getAssignorInstances(List<String> assignorClasses, Map<String, Object> configs) {
+        List<ConsumerPartitionAssignor> assignors = new ArrayList<>();
+
+        if (assignorClasses == null)
```

Comment:
[code dedup]: Can we leverage the refactored `getConfiguredInstance` in `AbstractConfig` by making it a public method?

Reply:
Not sure I follow? All `getConfiguredInstance` methods I see are public, but can't be used here as they expect the key and the class to instantiate it as. And of course, in this case we don't know ahead of time whether it will be PartitionAssignor or ConsumerPartitionAssignor?

I suppose we could try instantiating it as both and catch the KafkaException, but that doesn't seem much better

Reply:
I was referring to `private <T> T getConfiguredInstance(Object klass, Class<T> t, Map<String, Object> configPairs)`

The `Class<T> t` is only for type validation after the construction, but after some more thinking I'd agree with you that reusing this function would require us to move the check out of it and pushed to caller (and in this case the caller would be checking with two types) would not be worthy. Let's keep it as is.